cli/sleuth.ts

```diff
@@ -64,14 +65,19 @@ interface SolcOutput {
   errors?: string[],
 }
 
-function solcCompile(input: SolcInput): SolcOutput {
-  let solc;
-  try {
-    solc = require('solc');
-  } catch (e) {
-    throw new Error(`solc.js yarn dependency not found. Please build with optional dependencies included`);
+function solcCompile(
+  input: SolcInput,
+  compileFn: ((input: string) => string) | undefined
```

Comment:
I really think this should be async, which would be a way more common way to use it (e.g. via a WebWorker)